### PR TITLE
Example Terraform Code Optimization

### PR DESCRIPTION
**Pull request by marifse**
### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

**Comment by marifse** [comment]
@ocofaigh please check this PR, i have created after addressing the previous concerns. And if ok then please approve. Thanks

File: cra-config.yaml [hunk 1]
```diff
@@ -4,8 +4,6 @@ CRA_TARGETS:
   - CRA_TARGET: "examples/complete" # Target directory for CRA scan. If not provided, the CRA Scan will not be run.
     CRA_IGNORE_RULES_FILE: "cra-tf-validate-ignore-rules.json" # CRA Ignore file to use. If not provided, it checks the repo root directory for `cra-tf-validate-ignore-rules.json`
     PROFILE_ID: "0e6e7b5a-817d-4344-ab6f-e5d7a9c49520" # SCC profile ID (currently set to the FSCloud 1.4.0 profile).
-    # SCC_INSTANCE_ID: "" # The SCC instance ID to use to download profile for CRA scan. If not provided, a default global value will be used.
-    # SCC_REGION: "" # The IBM Cloud region that the SCC instance is in. If not provided, a default global value will be used.
-    # CRA_ENVIRONMENT_VARIABLES:  # An optional map of environment variables for CRA, where the key is the variable name and value is the value. Useful for providing TF_VARs.
-    #   TF_VAR_sample: "sample value"
-    #   TF_VAR_other:  "another value"
+    CRA_ENVIRONMENT_VARIABLES:  # An optional map of environment variables for CRA, where the key is the variable name and value is the value. Useful for providing TF_VARs.
+      TF_VAR_regions: "us-south"
```
Comment (ocofaigh) [inline cra-config.yaml:8]: typo. should be `TF_VAR_region`

**Review by ocofaigh — CHANGES_REQUESTED** [review]
the var name is wrong in cra-config.yaml

**Comment by marifse** [comment]
> the var name is wrong in cra-config.yaml

fixed.

**Review by ocofaigh — DISMISSED** [review]

**Comment by marifse** [comment]
@ocofaigh the variable value for prefix not found for unit test, whereas the value is already being passed as "options := setupOptions(t, "kms-key")".

**Comment by ocofaigh** [comment]
@marifse Since the test is using `TestOptionsDefault` (instead of `TestOptionsDefaultWithVars`) it means you need to explicitly pass any required variables to the test. So since region and prefix are now required, can you update the test to this:

```
func TestRunBasicExample(t *testing.T) {
	t.Parallel()

        options := testhelper.TestOptionsDefault(&testhelper.TestOptions{
		Testing:      t,
		TerraformDir: basicExampleTerraformDir,
		Prefix:       "kms-key-basic",
	})

        terraformVars := map[string]interface{}{
		"prefix": options.Prefix,
		"region":   options.Region,
		"existing_kms_instance_guid":   permanentResources["hpcs_south"],
	}

	options.TerraformVars = terraformVars

	output, err := options.RunTestConsistency()
	assert.Nil(t, err, "This should not have errored")
	assert.NotNil(t, output, "Expected some output")
}
```

**Comment by marifse** [comment]
@ocofaigh also for complete example test or only for this basic one?

**Comment by ocofaigh** [comment]
@marifse No the complete example is using `TestOptionsDefaultWithVars` which will always pass a value for prefix and region

**Comment by marifse** [comment]
> /run pipeline

please re-run pipeline, i have added the region value in basic example test

File: tests/pr_test.go [hunk 2]
```diff
@@ -59,12 +59,18 @@ func TestRunBasicExample(t *testing.T) {
...
-		TerraformVars: map[string]interface{}{
-			"existing_kms_instance_guid": permanentResources["hpcs_south"],
-		},
+		Prefix:       "kms-key-basic",
+		Region:       "us-south",
 	})
 
+	terraformVars := map[string]interface{}{
+		"prefix":                     options.Prefix,
+		"region":                     options.Region,
```
Comment (ocofaigh) [inline tests/pr_test.go:68]: actually the basic example doesnt have a region input var, so please remove this

File: tests/pr_test.go [hunk 3]
```diff
@@ -59,12 +59,18 @@ func TestRunBasicExample(t *testing.T) {
 	options := testhelper.TestOptionsDefault(&testhelper.TestOptions{
 		Testing:      t,
 		TerraformDir: basicExampleTerraformDir,
-		Prefix:       "hpcs",
-		TerraformVars: map[string]interface{}{
-			"existing_kms_instance_guid": permanentResources["hpcs_south"],
-		},
+		Prefix:       "kms-key-basic",
+		Region:       "us-south",
```
Comment (ocofaigh) [inline tests/pr_test.go:63]: same as https://github.com/terraform-ibm-modules/terraform-ibm-kms-key/pull/557/files#r1530357875

**Comment by marifse** [comment]
@ocofaigh region removed, please run the pipeline now.

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**